packages/utilities/src/diff/WebDiffManager.ts

```diff
@@ -0,0 +1,187 @@
+            if (doWeHaveGui === GuiResult.NO_GUI_NO_DISPLAY) {
+                errMsg += "\n\nIf you are running in an X Window environment," +
+                    "\nensure that your DISPLAY environment variable is set." +
+                    "\nFor example, type the following:" +
+                    "\n    echo $DISPLAY" +
+                    "\nIf it is not set, assign a valid value. For example:" +
+                    "\n    export DISPLAY=:0.0" +
+                    "\nThen try the --help-web option again.";
+
+                throw new ImperativeError({
+                    msg: errMsg
+                });
+            }
+
+            return;
+        }
+
+        if (this.checkWebDiffDirExists()) await new WebDiffGenerator(ImperativeConfig.instance, this.webDiffDir).buildDiffDir();
+
+        const htmlDiff = await html(patchDiff, {
+            outputFormat: "side-by-side",
+            matching: "lines",
+            diffStyle: "char",
+        });
+
+        if (htmlDiff != null) {
+            // writing the diff content into a textt file
+            fs.writeFileSync(path.join(this.webDiffDir, 'index.html'), this.genHtmlForDiffs(htmlDiff, patchDiff));
+            try {
+                ProcessUtils.openInDefaultApp(`file://${this.webDiffDir}/index.html`);
+            } catch (e) {
+                throw new ImperativeError({
+                    msg: "Failed to launch web diff, try running -h for console help instead",
+                    causeErrors: [e]
+                });
+            }
+        }
+    }
+
+
+    /**
+     * Check if the web diff direcory base has been generated
+     * at cli home
+     */
+    private async checkWebDiffDirExists() {
+        if (fs.existsSync(this.webDiffDir)) {
+            return true;
+        }
+        return false;
+    }
+
+
+    /**
+     * Gets the directory where built copy of web diff launcher is stored
+     * @readonly
+     * @private
+     * @returns {string} Absolute path of directory
+     */
+    private get webDiffDir(): string {
+        return path.join(ImperativeConfig.instance.cliHome, Constants.WEB_DIFF_DIR);
+    }
+
+    /**
+     * Returns header HTML for help page
+     * @private
+     * @param title - Title string for the page
+     */
+    private genHtmlForDiffs(htmlDiff: string, unifiedStringDiff: string): string {
+        return `<!DOCTYPE html>
+        <html>
+          <head>
+            <link
+              rel="stylesheet"
+              href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.7.1/styles/github.min.css"
+            />
+            <link
+              rel="stylesheet"
+              type="text/css"
+              href="https://cdn.jsdelivr.net/npm/diff2html/bundles/css/diff2html.min.css"
+            />
+            <script
+              type="text/javascript"
+              src="https://cdn.jsdelivr.net/npm/diff2html/bundles/js/diff2html-ui.min.js"
+            ></script>
```

Comment:
We may want to consider bundling these CSS and JS files with Imperative, so that diffs in the browser can work offline.

I don't want to unnecessarily expand the scope of this PR and think it's a really cool feature as-is, so I'd suggest its ok to create a separate issue to track making this work offline. 🙂

Reply:
yes, that's what we thought of, but for the future scope of the feature. Will do this thing with a new issue and PR.

Reply:
Sounds good 👍

Reply:
yes, that's what even if are thinking to do, but for the future scope of this features, as in some other issue and pr.  :slightly_smiling_face: